src/tcutility/analysis/vdd/manager.py

```diff
@@ -0,0 +1,220 @@
+PRINT_FORMAT = {"me": "%+.0f", "e": "%+.3f"}
+
+
+def change_unit_decorator(func):
+    @functools.wraps(func)
+    def wrapper(self, unit: str = "me", *args, **kwargs):
+        current_unit = self.unit
+        self.change_unit(unit)
+        result = func(self, *args, **kwargs)
+        self.change_unit(current_unit)
+        return result
+
+    return wrapper
+
+
+def create_vdd_charge_manager(results: result.Result, name: str = "VDDChargeManager") -> VDDChargeManager:
+    """Create a VDDChargeManager from a Result object."""
+    vdd_charges: Dict[str, List[charge.VDDCharge]] = {}
+    atom_symbols = results.molecule.atom_symbols  # type: ignore
+    frag_indices = results.molecule.frag_indices  # type: ignore
+
+    # Convert the VDD charges to VDDCharge objects
+    for irrep, charge_array in results.properties.vdd.items():  # type: ignore
+        irrep = "vdd" if irrep == "charges" else irrep
+        vdd_charges[irrep] = []
+        for atom_index, (frag_index, vdd_charge) in enumerate(zip(frag_indices, charge_array)):
+            atom_symbol = atom_symbols[atom_index]
+            vdd_charges[irrep].append(charge.VDDCharge(atom_index + 1, atom_symbol, vdd_charge, frag_index))
+
+    return VDDChargeManager(vdd_charges, name)
+
+
+@attrs.define
+class VDDChargeManager:
+    """Class to manage the VDD charges. It can be used to print the VDD charges in a nice table and write them to a text file or excel file."""
+
+    vdd_charges: Dict[str, List[charge.VDDCharge]]  # {"vdd": [VDDCharge, ...], "irrep1": [VDDCharge, ...], ...}
+    name: str = "VDDChargeManager"  # name of the VDDChargeManager
+    unit: str = attrs.field(init=False, default="e")  # unit of the VDD charges. Available units are "me" (mili-electrons) and "e" (electrons)
+    irreps: List[str] = attrs.field(init=False, default=set())
+
+    def __attrs_post_init__(self):
+        self.irreps = list(self.vdd_charges.keys())
+        self.change_unit("me")
+
+    def __str__(self) -> str:
+        """Prints the VDD charges in a nice table. Checks if the calculation is a fragment calculation and prints the summed VDD charges if it is."""
+        individual_charges_table = self.get_vdd_charges_table(self.unit)
+        summed_charges_table = self.get_summed_vdd_charges_table(self.unit)
+
+        ret_str = f"{self.name}\nVDD charges (in unit {self.unit}):\n{individual_charges_table}\n\n"
+        if self.is_fragment_calculation:
+            ret_str += f"Summed VDD charges (in unit {self.unit}):\n{summed_charges_table}\n"
+
+        return ret_str
+
+    @property
+    def is_fragment_calculation(self) -> bool:
+        """Check if the calculation is a fragment calculation by checking if the highest fragment index is equal to the number of atoms."""
+        return max([charge.frag_index for charges in self.vdd_charges.values() for charge in charges]) != len(self.vdd_charges["vdd"])
+
+    def charge_is_conserved(self, mol_charge: int) -> bool:
```

Comment:
I prefer to have the charge manager as a standalone class that takes input of the keys present in the result instance and has not connection with the results instance thereafter. This is to decouple the manager so that testing becomes easier and you don't risk any unfavored side-effects when changing the results class or the manager class.